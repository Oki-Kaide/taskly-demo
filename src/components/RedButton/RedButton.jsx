import React from 'react';
import PropTypes from 'prop-types';
import './RedButton.sass';

const RedButton = ({text, text2, handleOnClick, className, isLoading}) => {
  let buttonContent;
  if (isLoading) {
    buttonContent = (<div className="sk-fading-circle">
    <div className="sk-circle1 sk-circle"></div>
    <div className="sk-circle2 sk-circle"></div>
    <div className="sk-circle3 sk-circle"></div>
    <div className="sk-circle4 sk-circle"></div>
    <div className="sk-circle5 sk-circle"></div>
    <div className="sk-circle6 sk-circle"></div>
    <div className="sk-circle7 sk-circle"></div>
    <div className="sk-circle8 sk-circle"></div>
    <div className="sk-circle9 sk-circle"></div>
    <div className="sk-circle10 sk-circle"></div>
    <div className="sk-circle11 sk-circle"></div>
    <div className="sk-circle12 sk-circle"></div>
  </div>)
  } else {
    if (text2) {
      buttonContent = (<><span>{text}</span><span>{text2}</span></>)
    } else {
      buttonContent = (<span>{text}</span>);
    }
  }
  return (
    <button className={`red-button ${className}`} type="button" onClick={handleOnClick}>
      { buttonContent }
    </button>
  );
}

RedButton.propTypes = {
  text: PropTypes.string.isRequired,
  text2: PropTypes.string,
  handleOnClick: PropTypes.func,
  isLoading: PropTypes.bool,
};

RedButton.defaultProps = {
  handleOnClick: () => {},
  isLoading: false,
}

export default RedButton;
