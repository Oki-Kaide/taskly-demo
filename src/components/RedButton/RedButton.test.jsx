import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RedButton from './RedButton';

const createTestProps = props => ({
  text: 'button text',
  ...props,
});

describe('RedButton component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<RedButton {...props} />);
  });

  it('renders without crashing while loading', () => {
    shallow(<RedButton {...props} isLoading={true} />);
  });

  it('renders without crashing with a second text', () => {
    shallow(<RedButton {...props} text2='hi' />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<RedButton {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('tests click', () => {
    const component = shallow(<RedButton {...props} />);
    component.find('button').simulate('click');
  });
});
