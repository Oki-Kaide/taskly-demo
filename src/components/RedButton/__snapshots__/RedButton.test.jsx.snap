// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedButton component matches the rendered snapshot 1`] = `
<button
  className="red-button undefined"
  onClick={[Function]}
  type="button"
>
  <span>
    button text
  </span>
</button>
`;
